[Failing Test]: BigtableReadSchemaTransformProviderIT. testRead flaky

### What happened?

https://ci-beam.apache.org/job/beam_PostCommit_Java_DataflowV2/3227/testReport/junit/org.apache.beam.sdk.io.gcp.bigtable/BigtableReadSchemaTransformProviderIT/testRead/

Error message:

```
java.lang.RuntimeException: org.apache.beam.sdk.util.UserCodeException: com.google.api.gax.rpc.NotFoundException: io.grpc.StatusRuntimeException: NOT_FOUND: Table not found: projects/apache-beam-testing/instances/beam-test/tables/BigtableReadSchemaTransformIT
	at org.apache.beam.sdk.util.UserCodeException.wrap(UserCodeException.java:39)
	at org.apache.beam.sdk.io.Read$BoundedSourceAsSDFWrapperFn$DoFnInvoker.invokeSplitRestriction(Unknown Source)
...
Caused by: com.google.api.gax.rpc.NotFoundException: io.grpc.StatusRuntimeException: NOT_FOUND: Table not found: projects/apache-beam-testing/instances/beam-test/tables/BigtableReadSchemaTransformIT
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:90)
...
Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
		at com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57)
		at com.google.cloud.bigtable.data.v2.BigtableDataClient.sampleRowKeys(BigtableDataClient.java:870)
		at org.apache.beam.sdk.io.gcp.bigtable.BigtableServiceImpl.getSampleRowKeys(BigtableServiceImpl.java:637)
		at org.apache.beam.sdk.io.gcp.bigtable.BigtableIO$BigtableSource.getSampleRowKeys(BigtableIO.java:1316)
Caused by: io.grpc.StatusRuntimeException: NOT_FOUND: Table not found: projects/apache-beam-testing/instances/beam-test/tables/BigtableReadSchemaTransformIT
	at io.grpc.Status.asRuntimeException(Status.java:539)
	at com.google.api.gax.grpc.GrpcDirectStreamController$ResponseObserverAdapter.onClose(GrpcDirectStreamController.java:148)
...
org.apache.beam.sdk.util.UserCodeException: com.google.api.gax.rpc.NotFoundException: io.grpc.StatusRuntimeException: NOT_FOUND: Table not found: projects/apache-beam-testing/instances/beam-test/tables/BigtableReadSchemaTransformIT
	at org.apache.beam.sdk.util.UserCodeException.wrap(UserCodeException.java:39)
	at org.apache.beam.sdk.io.Read$BoundedSourceAsSDFWrapperFn$DoFnInvoker.invokeSplitRestriction(Unknown Source)
	at org.apache.beam.fn.harness.FnApiDoFnRunner.processElementForSplitRestriction(FnApiDoFnRunner.java:887)
...
Caused by: com.google.api.gax.rpc.NotFoundException: io.grpc.StatusRuntimeException: NOT_FOUND: Table not found: projects/apache-beam-testing/instances/beam-test/tables/BigtableReadSchemaTransformIT
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:90)
...
```

Likely some race condition?

### Issue Failure

Failure: Test is flaky

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProviderIT.java
+++ b/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProviderIT.java
@@ -32,6 +32,7 @@ import java.nio.ByteBuffer;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Date;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
@@ -56,7 +57,7 @@ public class BigtableReadSchemaTransformProviderIT {
 
   private static final String COLUMN_FAMILY_NAME_1 = "test_cf_1";
   private static final String COLUMN_FAMILY_NAME_2 = "test_cf_2";
-  private BigtableTableAdminClient adminClient;
+  private BigtableTableAdminClient tableAdminClient;
   private BigtableDataClient dataClient;
   private String tableId;
   private String projectId;
@@ -93,7 +94,7 @@ public class BigtableReadSchemaTransformProviderIT {
   public void setup() throws Exception {
     BigtableTestOptions options =
         TestPipeline.testingPipelineOptions().as(BigtableTestOptions.class);
-    projectId = options.as(GcpOptions.class).getProject();
+    projectId = "google.com:clouddfe";//options.as(GcpOptions.class).getProject();
     instanceId = options.getInstanceId();
 
     BigtableDataSettings settings =
@@ -106,31 +107,31 @@ public class BigtableReadSchemaTransformProviderIT {
             .setProjectId(projectId)
             .setInstanceId(instanceId)
             .build();
-    adminClient = BigtableTableAdminClient.create(adminSettings);
+    tableAdminClient = BigtableTableAdminClient.create(adminSettings);
+
+    tableId = String.format("BTReadSchemaTransformIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date());
+    if (!tableAdminClient.exists(tableId)) {
+      CreateTableRequest createTableRequest =
+              CreateTableRequest.of(tableId)
+                      .addFamily(COLUMN_FAMILY_NAME_1)
+                      .addFamily(COLUMN_FAMILY_NAME_2);
+      tableAdminClient.createTable(createTableRequest);
+    }
   }
 
   @After
   public void tearDown() {
     try {
-      adminClient.deleteTable(tableId);
+      tableAdminClient.deleteTable(tableId);
       System.out.printf("Table %s deleted successfully%n", tableId);
     } catch (NotFoundException e) {
       System.err.println("Failed to delete a non-existent table: " + e.getMessage());
     }
     dataClient.close();
-    adminClient.close();
+    tableAdminClient.close();
   }
 
-  public List<Row> writeToTable(int numRows) throws Exception {
-    // Checks if table exists, creates table if does not exist.
-    if (!adminClient.exists(tableId)) {
-      CreateTableRequest createTableRequest =
-          CreateTableRequest.of(tableId)
-              .addFamily(COLUMN_FAMILY_NAME_1)
-              .addFamily(COLUMN_FAMILY_NAME_2);
-      adminClient.createTable(createTableRequest);
-    }
-
+  public List<Row> writeToTable(int numRows) {
     List<Row> expectedRows = new ArrayList<>();
 
     try {
@@ -199,6 +200,7 @@ public class BigtableReadSchemaTransformProviderIT {
 
         expectedRows.add(expectedRow);
       }
+      System.out.printf("Finished writing %s rows to table %s", numRows, tableId);
     } catch (NotFoundException e) {
       throw new RuntimeException("Failed to write to table", e);
     }
@@ -206,8 +208,7 @@ public class BigtableReadSchemaTransformProviderIT {
   }
 
   @Test
-  public void testRead() throws Exception {
-    tableId = "BigtableReadSchemaTransformIT";
+  public void testRead() {
     List<Row> expectedRows = writeToTable(20);
 
     BigtableReadSchemaTransformConfiguration config =

--- a/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProviderIT.java
+++ b/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProviderIT.java
@@ -50,9 +50,14 @@ import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @RunWith(JUnit4.class)
 public class BigtableReadSchemaTransformProviderIT {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(BigtableReadSchemaTransformProviderIT.class);
+
   @Rule public final transient TestPipeline p = TestPipeline.create();
 
   private static final String COLUMN_FAMILY_NAME_1 = "test_cf_1";
@@ -123,9 +128,9 @@ public class BigtableReadSchemaTransformProviderIT {
   public void tearDown() {
     try {
       tableAdminClient.deleteTable(tableId);
-      System.out.printf("Table %s deleted successfully%n", tableId);
+      LOG.info("Table %s deleted successfully.", tableId);
     } catch (NotFoundException e) {
-      System.err.println("Failed to delete a non-existent table: " + e.getMessage());
+      LOG.warn("Failed to delete a non-existent table [{}]: \n{}", tableId, e.getMessage());
     }
     dataClient.close();
     tableAdminClient.close();
@@ -200,7 +205,7 @@ public class BigtableReadSchemaTransformProviderIT {
 
         expectedRows.add(expectedRow);
       }
-      System.out.printf("Finished writing %s rows to table %s", numRows, tableId);
+      LOG.info("Finished writing {} rows to table {}", numRows, tableId);
     } catch (NotFoundException e) {
       throw new RuntimeException("Failed to write to table", e);
     }

--- a/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProviderIT.java
+++ b/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProviderIT.java
@@ -128,7 +128,7 @@ public class BigtableReadSchemaTransformProviderIT {
   public void tearDown() {
     try {
       tableAdminClient.deleteTable(tableId);
-      LOG.info("Table %s deleted successfully.", tableId);
+      LOG.info("Table {} deleted successfully.", tableId);
     } catch (NotFoundException e) {
       LOG.warn("Failed to delete a non-existent table [{}]: \n{}", tableId, e.getMessage());
     }

--- a/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProviderIT.java
+++ b/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProviderIT.java
@@ -94,7 +94,7 @@ public class BigtableReadSchemaTransformProviderIT {
   public void setup() throws Exception {
     BigtableTestOptions options =
         TestPipeline.testingPipelineOptions().as(BigtableTestOptions.class);
-    projectId = "google.com:clouddfe";//options.as(GcpOptions.class).getProject();
+    projectId = options.as(GcpOptions.class).getProject();
     instanceId = options.getInstanceId();
 
     BigtableDataSettings settings =
@@ -112,9 +112,9 @@ public class BigtableReadSchemaTransformProviderIT {
     tableId = String.format("BTReadSchemaTransformIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date());
     if (!tableAdminClient.exists(tableId)) {
       CreateTableRequest createTableRequest =
-              CreateTableRequest.of(tableId)
-                      .addFamily(COLUMN_FAMILY_NAME_1)
-                      .addFamily(COLUMN_FAMILY_NAME_2);
+          CreateTableRequest.of(tableId)
+              .addFamily(COLUMN_FAMILY_NAME_1)
+              .addFamily(COLUMN_FAMILY_NAME_2);
       tableAdminClient.createTable(createTableRequest);
     }
   }

